Change docstring to Google format

Narann opened the pull request:
As requested in https://github.com/sqlboy/fileseq/issues/68, docstring format is now Google one.

The current documentation config rely on restructured text. So the generated doc is not that nice. What can I do in this regard?

Unit tests pass except the ones using what seems to be an absolute path.

justinfx commented on src/fileseq/filesequence.py:
```diff
@@ -198,36 +212,42 @@ def frameSet(self):
-        :param ext: the new file extension
-        :rtype: None
+        Args:
+            ext (str):
```
Forgot to preserve the description text for the "ext" field

justinfx commented on src/fileseq/filesequence.py:
```diff
@@ -198,36 +212,42 @@ def frameSet(self):
+            ext (str):
+
+        Returns:
+            None:
```
This isn't needed if nothing is being returned

justinfx commented on src/fileseq/filesequence.py:
```diff
@@ -257,8 +281,9 @@ def setFrameRange(self, frange):
-        :param frange: a properly formatted frame range, as per :class:`fileseq.frameset.FrameSet`
-        :rtype: None
+        Args:
+            frange: a properly formatted frame range, as per
```
Missing type:  `frange (str)`

justinfx commented on src/fileseq/frameset.py:
```diff
@@ -74,17 +67,31 @@ def __new__(cls, *args, **kwargs):
-        :raises: :class:`fileseq.exceptions.ParseException` if the frame range
-                 (or a portion of it) could not be parsed
+        Args:
+            frange (str or FrameSet): the frame range as a string (ie "1-100x5")
```
Missing rst formatting:  
```
:class:`FrameSet`
```

justinfx commented on src/fileseq/frameset.py:
```diff
@@ -230,9 +244,15 @@ def _cast_to_frameset(cls, other):
-        :rtype: :class:`FrameSet`
-        :returns: :class:`NotImplemented` if a comparison is impossible
+        Args:
+            other: the :class:`FrameSet`, set, frozenset, or iterable to be
```
Should be:

```
other (:class:`FrameSet` or set or frozenset or or iterable): item to be compared 
```

justinfx commented on src/fileseq/filesequence.py:
```diff
@@ -198,36 +212,42 @@ def frameSet(self):
-        :rtype: None
+        Args:
+            frameSet (:class:`fileseq.frameset.FrameSet`): the new
+            :class:`fileseq.frameset.FrameSet` object
```
Sphinx complains about this missing a newline after the last Args field

justinfx commented:
For the sphinx-build to process the google docstrings correctly you need to update the docs/conf.py:

```
extensions = [
    'sphinx.ext.autodoc',
    'sphinx.ext.viewcode',
    'sphinx.ext.napoleon',  # <---
]
```

Also, I would recommend not moving the class level documentation into the `__init__` methods. Sphinx will not document them by default unless you set `napoleon_include_init_with_doc = True`. But even then, the methods are all sorted so the constructor docs get shown much further down. Its cleaner to just have them all in the class docstring.

justinfx commented:
Also I forgot to say.... Thanks for this work!

Narann commented:
I've fixed all what you pointed, moved the init doc string to the class one.

I've also update some domains to have proper link in generated doc. I'm sure there must be a way to avoid full namespaces and have `FrameSet` instead of `fileseq.frameset.FrameSet` but I didn't find how.

Hope it's OK. :)

justinfx commented:
Thanks for the fixes.
The way to do relative references would be `.FrameSet`

Narann commented:
> The way to do relative references would be `.FrameSet`

Perfect, let me fix this ! :)

Narann commented:
Dopc looks nicer now.

I realize there is some docstring calling an unknown `FrameRange` class.

justinfx commented:
I fixed up those old docstring refs. Thanks again!
